azure_monitor/src/azure_monitor/trace.py

```diff
@@ -104,16 +105,18 @@ def span_to_envelope(self, span):  # noqa pylint: disable=too-many-branches
                 status_code = span.attributes["http.status_code"]
                 data.responseCode = str(status_code)
                 data.success = 200 <= status_code < 400
+            elif span.status == StatusCanonicalCode.OK:
+                data.success = True
         else:
             envelope.name = "Microsoft.ApplicationInsights.RemoteDependency"
             data = protocol.RemoteDependency(
                 name=span.name,
-                id="|{:032x}.{:016x}.".format(
-                    span.context.trace_id, span.context.span_id
+                id="{:016x}".format(
+                    span.context.span_id
                 ),
-                resultCode="0",  # TODO
+                resultCode="0",
```

Comment:
can we put canonical code here?

Reply:
Do you mean to use the "StatusCanonicalCode"?

Reply:
yes